lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java

```diff
@@ -127,26 +134,39 @@ public ConjureStartTransactionsResponse startTransactions(ConjureStartTransactio
             ToLongFunction<T> lowerBoundExtractor,
             ToLongFunction<T> upperBoundExtractor,
             OperationType operationType) {
+        // take snapshot before making the request
+        Instant wallClockTimeBeforeRequest = Instant.now();
```

Comment:
Probably want to take the time snapshot after we get the bounds? Awkward if the bounds update in between

Reply:
This is also happening but I can make it immediately after we get the response.